solidity/ExpirableAirdrop.sol

```diff
@@ -1,60 +1,108 @@
   /// @param _token token to airdrop
+  /// @param _expirationTimestamp expiration timestamp
   /// @param _merkleRoot of claimees
   constructor(
+    address _governor,
     IERC20 _token,
     uint256 _expirationTimestamp,
     bytes32 _merkleRoot
-  ) {
+  ) Governable(_governor) {
     token = _token;
     expirationTimestamp = _expirationTimestamp;
     merkleRoot = _merkleRoot;
   }
 
-  /// @notice Allows claiming tokens if address is part of merkle tree
-  /// @param to address of claimee
-  /// @param amount of tokens owed to claimee
-  /// @param proof merkle proof to prove address and amount are in tree
-  function claim(
-    address to,
+  function _claim(
+    address claimee,
+    address receiver,
     uint256 amount,
     bytes32[] calldata proof
-  ) external {
+  ) internal {
     // CHECKS
     if (block.timestamp > expirationTimestamp) revert Expired();
-    if (hasClaimed[to]) revert AlreadyClaimed();
-    bytes32 leaf = keccak256(abi.encodePacked(to, amount));
+    if (hasClaimed[claimee]) revert AlreadyClaimed();
+    bytes32 leaf = keccak256(abi.encodePacked(claimee, amount));
     bool isValidLeaf = MerkleProof.verify(proof, merkleRoot, leaf);
     if (!isValidLeaf) revert NotInMerkle();
 
     // EFFECTS
-    hasClaimed[to] = true;
+    hasClaimed[claimee] = true;
+
+    // INTERACTIONS
+    token.transfer(receiver, amount);
+
+    emit Claimed(claimee, receiver, amount);
+  }
+
+  /// @notice Allows claiming tokens, if address is part of merkle tree
+  /// @param claimee address of claimee
+  /// @param amount of tokens owed to claimee
+  /// @param proof merkle proof to prove address and amount are in tree
+  function claim(
+    address claimee,
+    uint256 amount,
+    bytes32[] calldata proof
+  ) external {
+    _claim(claimee, claimee, amount, proof);
+  }
+
+  /// @notice Allows claiming tokens and send tokens to a receiver, if address is part of merkle tree
+  /// @param receiver address to send tokens
+  /// @param amount of tokens owed to claimee
+  /// @param proof merkle proof to prove address and amount are in tree
+  function claimAndTransfer(
+    address receiver,
+    uint256 amount,
+    bytes32[] calldata proof
+  ) external {
+    _claim(msg.sender, receiver, amount, proof);
+  }
+
+  /// @notice Deposit tokens to airdrop
+  /// @param amount of tokens to deposit
+  function depositTokens(uint256 amount) external {
+    // INTERACTIONS
+    token.transferFrom(msg.sender, address(this), amount);
+
+    emit Deposited(amount);
+  }
+
+  /// @notice Return unclaimed tokens to governor
+  function retrieveTokens() external {
+    // CHECKS
+    if (block.timestamp <= expirationTimestamp) revert NotExpired();
 
     // INTERACTIONS
-    token.transfer(to, amount);
+    address governor = IGovernable(address(this)).governor();
+    token.transfer(governor, token.balanceOf(address(this)));
 
-    emit Claimed(to, amount);
+    emit Retrieved(block.timestamp, governor);
```

Comment:
We don't need to emit the block's timestamp, that is already calculable by looking at the transaction, and it costs gas to emit it